Fail to build with CATKIN_ENABLE_TESTING=0

From [the ROS  documentation](https://docs.ros.org/en/jade/api/catkin/html/howto/format2/index.html#configuring-and-running-unit-tests), test configurations should be optional and skipped with the cmake arg: `-DCATKIN_ENABLE_TESTING=0`.

When building `rosfmt` with this argument we get a cmake error about tests:

```
docker run --rm ros:noetic bash -c "mkdir -p ws/src && cd ws/src && apt update && apt install -y git python3-catkin-tools && git clone https://github.com/xqms/rosfmt.git && cd .. && catkin build -DCATKIN_ENABLE_TESTING=0"

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Get:1 http://packages.ros.org/ros/ubuntu focal InRelease [4679 B]
Get:2 http://packages.ros.org/ros/ubuntu focal/main amd64 Packages [774 kB]
Get:3 http://security.ubuntu.com/ubuntu focal-security InRelease [114 kB]
Get:4 http://security.ubuntu.com/ubuntu focal-security/universe amd64 Packages [1047 kB]
Get:5 http://security.ubuntu.com/ubuntu focal-security/multiverse amd64 Packages [28.5 kB]
Get:6 http://security.ubuntu.com/ubuntu focal-security/main amd64 Packages [2681 kB]
Get:7 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages [2203 kB]
Get:8 http://archive.ubuntu.com/ubuntu focal InRelease [265 kB]
Get:9 http://archive.ubuntu.com/ubuntu focal-updates InRelease [114 kB]
Get:10 http://archive.ubuntu.com/ubuntu focal-backports InRelease [108 kB]
Get:11 http://archive.ubuntu.com/ubuntu focal/main amd64 Packages [1275 kB]
Get:12 http://archive.ubuntu.com/ubuntu focal/restricted amd64 Packages [33.4 kB]
Get:13 http://archive.ubuntu.com/ubuntu focal/multiverse amd64 Packages [177 kB]
Get:14 http://archive.ubuntu.com/ubuntu focal/universe amd64 Packages [11.3 MB]
Get:15 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 Packages [1343 kB]
Get:16 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 Packages [31.2 kB]
Get:17 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [3169 kB]
Get:18 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages [2366 kB]
Get:19 http://archive.ubuntu.com/ubuntu focal-backports/main amd64 Packages [55.2 kB]
Get:20 http://archive.ubuntu.com/ubuntu focal-backports/universe amd64 Packages [28.6 kB]
Fetched 27.2 MB in 1min 23s (326 kB/s)
Reading package lists...
Building dependency tree...
Reading state information...
10 packages can be upgraded. Run 'apt list --upgradable' to see them.

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  git-man less libcurl3-gnutls liberror-perl python3-importlib-metadata
  python3-more-itertools python3-osrf-pycommon python3-zipp
Suggested packages:
  gettext-base git-daemon-run | git-daemon-sysvinit git-doc git-el git-email
  git-gui gitk gitweb git-cvs git-mediawiki git-svn
The following NEW packages will be installed:
  git git-man less libcurl3-gnutls liberror-perl python3-catkin-tools
  python3-importlib-metadata python3-more-itertools python3-osrf-pycommon
  python3-zipp
0 upgraded, 10 newly installed, 0 to remove and 10 not upgraded.
Need to get 6110 kB of archives.
After this operation, 40.8 MB of additional disk space will be used.
Get:1 http://packages.ros.org/ros/ubuntu focal/main amd64 python3-osrf-pycommon all 2.0.2-1 [23.3 kB]
Get:2 http://packages.ros.org/ros/ubuntu focal/main amd64 python3-catkin-tools all 0.9.2-1 [158 kB]
Get:3 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 less amd64 551-1ubuntu0.1 [123 kB]
Get:4 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libcurl3-gnutls amd64 7.68.0-1ubuntu2.18 [233 kB]
Get:5 http://archive.ubuntu.com/ubuntu focal/main amd64 liberror-perl all 0.17029-1 [26.5 kB]
Get:6 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 git-man all 1:2.25.1-1ubuntu3.11 [887 kB]
Get:7 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 git amd64 1:2.25.1-1ubuntu3.11 [4605 kB]
Get:8 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-more-itertools all 4.2.0-1build1 [39.4 kB]
Get:9 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-zipp all 1.0.0-1 [5312 B]
Get:10 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-importlib-metadata all 1.5.0-1 [9992 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 6110 kB in 14s (439 kB/s)
Selecting previously unselected package less.
(Reading database ... 45871 files and directories currently installed.)
Preparing to unpack .../0-less_551-1ubuntu0.1_amd64.deb ...
Unpacking less (551-1ubuntu0.1) ...
Selecting previously unselected package libcurl3-gnutls:amd64.
Preparing to unpack .../1-libcurl3-gnutls_7.68.0-1ubuntu2.18_amd64.deb ...
Unpacking libcurl3-gnutls:amd64 (7.68.0-1ubuntu2.18) ...
Selecting previously unselected package liberror-perl.
Preparing to unpack .../2-liberror-perl_0.17029-1_all.deb ...
Unpacking liberror-perl (0.17029-1) ...
Selecting previously unselected package git-man.
Preparing to unpack .../3-git-man_1%3a2.25.1-1ubuntu3.11_all.deb ...
Unpacking git-man (1:2.25.1-1ubuntu3.11) ...
Selecting previously unselected package git.
Preparing to unpack .../4-git_1%3a2.25.1-1ubuntu3.11_amd64.deb ...
Unpacking git (1:2.25.1-1ubuntu3.11) ...
Selecting previously unselected package python3-more-itertools.
Preparing to unpack .../5-python3-more-itertools_4.2.0-1build1_all.deb ...
Unpacking python3-more-itertools (4.2.0-1build1) ...
Selecting previously unselected package python3-zipp.
Preparing to unpack .../6-python3-zipp_1.0.0-1_all.deb ...
Unpacking python3-zipp (1.0.0-1) ...
Selecting previously unselected package python3-importlib-metadata.
Preparing to unpack .../7-python3-importlib-metadata_1.5.0-1_all.deb ...
Unpacking python3-importlib-metadata (1.5.0-1) ...
Selecting previously unselected package python3-osrf-pycommon.
Preparing to unpack .../8-python3-osrf-pycommon_2.0.2-1_all.deb ...
Unpacking python3-osrf-pycommon (2.0.2-1) ...
Selecting previously unselected package python3-catkin-tools.
Preparing to unpack .../9-python3-catkin-tools_0.9.2-1_all.deb ...
Unpacking python3-catkin-tools (0.9.2-1) ...
Setting up python3-more-itertools (4.2.0-1build1) ...
Setting up less (551-1ubuntu0.1) ...
Setting up libcurl3-gnutls:amd64 (7.68.0-1ubuntu2.18) ...
Setting up python3-zipp (1.0.0-1) ...
Setting up liberror-perl (0.17029-1) ...
Setting up git-man (1:2.25.1-1ubuntu3.11) ...
Setting up python3-importlib-metadata (1.5.0-1) ...
Setting up git (1:2.25.1-1ubuntu3.11) ...
Setting up python3-osrf-pycommon (2.0.2-1) ...
Setting up python3-catkin-tools (0.9.2-1) ...
Processing triggers for libc-bin (2.31-0ubuntu9.9) ...
Processing triggers for mime-support (3.64ubuntu1) ...
Cloning into 'rosfmt'...
------------------------------------------------------
Profile:                     default
Extending:             [env] /opt/ros/noetic
Workspace:                   /ws
------------------------------------------------------
Build Space:        [exists] /ws/build
Devel Space:        [exists] /ws/devel
Install Space:      [unused] /ws/install
Log Space:         [missing] /ws/logs
Source Space:       [exists] /ws/src
DESTDIR:            [unused] None
------------------------------------------------------
Devel Space Layout:          linked
Install Space Layout:        None
------------------------------------------------------
Additional CMake Args:       -DCATKIN_ENABLE_TESTING=0
Additional Make Args:        None
Additional catkin Make Args: None
Internal Make Job Server:    True
Cache Job Environments:      False
------------------------------------------------------
Buildlisted Packages:        None
Skiplisted Packages:         None
------------------------------------------------------
Workspace configuration appears valid.

NOTE: Forcing CMake to run for each package.
------------------------------------------------------
[build] Found 1 packages in 0.0 seconds.                                       
[build] Updating package table.                                                
Starting >>> catkin_tools_prebuild                                             
Finished <<< catkin_tools_prebuild                [ 1.4 seconds ]              
Starting >>> rosfmt                                                            
Failed <<< rosfmt                               [ 1.7 seconds ]                
[build] Summary: 1 of 2 packages succeeded.                                    
[build] Ignored: None.                                                         
[build] Warnings: None.                                                        
[build] Abandoned: No packages were abandoned.                                 
[build] Failed: 1 packages failed.                                             
[build] Runtime: 3.1 seconds total.                                            
[build] Note: Workspace packages have changed, please re-source setup files to use them.
_______________________________________________________________________________
Errors << rosfmt:cmake /ws/logs/rosfmt/build.cmake.000.log                     
CMake Error at /ws/src/rosfmt/CMakeLists.txt:90 (add_dependencies):
  Cannot add target-level dependencies to non-existent target "run_tests".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.


cd /ws/build/rosfmt; catkin build --get-env rosfmt | catkin env -si  /usr/bin/cmake /ws/src/rosfmt --no-warn-unused-cli -DCATKIN_DEVEL_PREFIX=/ws/devel/.private/rosfmt -DCMAKE_INSTALL_PREFIX=/ws/install -DCATKIN_ENABLE_TESTING=0; cd -

...............................................................................
Failed << rosfmt:cmake                          [ Exited with code 1 ]
```

Making the tests part conditional allows to build properly.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,19 +76,20 @@ target_link_libraries(rosfmt9
 )
 add_dependencies(rosfmt9 fmt fmt_headers)
 
-add_executable(simple_test
-	test/test.cpp
-)
-target_link_libraries(simple_test rosfmt9)
-add_test(
-	NAME SimpleTest
-	COMMAND simple_test
-)
-add_custom_target(run_simple_test
-	COMMAND simple_test
-)
-add_dependencies(run_tests run_simple_test)
-
+if(CATKIN_ENABLE_TESTING)
+	add_executable(simple_test
+		test/test.cpp
+	)
+	target_link_libraries(simple_test rosfmt9)
+	add_test(
+		NAME SimpleTest
+		COMMAND simple_test
+	)
+	add_custom_target(run_simple_test
+		COMMAND simple_test
+	)
+	add_dependencies(run_tests run_simple_test)
+endif()
 
 install(
 	DIRECTORY ${fmt_HEADER_DEST}/fmt/

